plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/ElasticsearchClient.java

```diff
@@ -145,12 +141,12 @@
     @Inject
     public ElasticsearchClient(
             ElasticsearchConfig config,
-            Optional<AwsSecurityConfig> awsSecurityConfig,
+            @ForElasticsearchSigner Optional<HttpRequestInterceptor> awsRequestSigner,
```

Comment:
ElasticsearchClient should not expose any details in its about the fact that it uses Apache HTTP client internally.

Reply:
Changed it to inject `Optional<AwsSignerCredentialsProvider>` instead.